vm/systemSmartContracts/delegation.go

```diff
@@ -691,21 +734,27 @@ func (d *delegation) unStakeNodes(args *vmcommon.ContractCallInput) vmcommon.Ret
 }
 
 func (d *delegation) unBondNodes(args *vmcommon.ContractCallInput) vmcommon.ReturnCode {
-	returnCode := d.checkOwnerCallValueGas(args)
+	returnCode := d.checkOwnerCallValueGasAndDuplicates(args)
 	if returnCode != vmcommon.Ok {
 		return returnCode
 	}
 	if len(args.Arguments) == 0 {
 		d.eei.AddReturnMessage("not enough arguments")
 		return vmcommon.FunctionWrongSignature
 	}
+	//TODO: Shouldn't be checked if there is enough gas for all the nodes which would be unBonded?
+	err := d.eei.UseGas(uint64(len(args.Arguments)) * d.gasCost.MetaChainSystemSCsCost.DelegationOps)
```

Comment:
no need for further checks. all good.

Reply:
But this call is missing at all in original branch. We should keep this check added or not?